### PR TITLE
Snake game

### PR DESCRIPTION
**Pull request by atharva012004**
## Related Issue
#232 

## Email id used to register for VSoc'24

atharavashihurkar@gmail.com

## Description

Enhance the basic snake game to an advanced level by adding the following features:
1. **Levels and Increasing Difficulty**:
   - Introduce levels where the game speed increases or new obstacles appear.
   - Add different types of food with varying points and effects (e.g., bonus food that appears for a short time).

2. **Obstacles**:
   - Add static or moving obstacles that the snake must avoid.
   - Implement different obstacle types with varying difficulty.

3. **Power-ups**:
   - Include power-ups that temporarily enhance the snake's abilities (e.g., speed boost, invincibility).
   - Add a countdown timer for the power-ups' effects.

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [X] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

https://github.com/dhairyagothi/50_days_50_web_project/assets/136626840/06ee3022-e576-4992-816d-87eea00970de

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Additional Context

Updating the game to an advanced level will significantly enhance user engagement and replay value. Adding levels, obstacles, and power-ups will make the game more challenging and enjoyable.

### CODE REVIEW

**Comment by dhairyagothi** [comment]
@atharva012004 make snake move faster

File: public/snake_game/script.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+let gameOver = false;
+let foodX, foodY, bonusFoodX, bonusFoodY, powerUpX, powerUpY;
+let bonusFoodVisible = false, powerUpVisible = false;
+let snakeX = 5, snakeY = 5;
+let velocityX = 0, velocityY = 0;
+let snakeBody = [];
+let setIntervalId, powerUpTimerId;
+let score = 0;
+let level = 1;
+let gameSpeed = 300;
```
Comment (dhairyagothi) [inline public/snake_game/script.js:18]: increase this

**Review by dhairyagothi — CHANGES_REQUESTED** [review]

**Comment by atharva012004** [comment]
The snake will move fast as the level will increase there are total 29 levels

**Comment by dhairyagothi** [comment]
increase some at level 1 also bro

**Comment by dhairyagothi** [comment]
okay np

**Comment by atharva012004** [comment]
Have you tried once actually it is lagging in recording review once then if you want i can try that

**Pull request merged**